src/scripts/install_machine.py

```diff
@@ -322,10 +305,15 @@ def install_vm(args):
 
 
 def inject_metadata(xml):
+    # Register used namespaces
+    ns = {"cockpit_machines": "https://github.com/cockpit-project/cockpit-machines"}
+    ET.register_namespace("cockpit_machines", ns["cockpit_machines"])
+    ET.register_namespace("libosinfo", "http://libosinfo.org/xmlns/libvirt/domain/1.0")
+
     # ET.fromstring() already wants UTF-8 encoded bytes
     root = ET.fromstring(xml)
     metadata = root.find('metadata')
-    cockpit_machines_metadata = metadata.find('cockpit_machines')
+    cockpit_machines_metadata = metadata.find('cockpit_machines:data', ns)
     if cockpit_machines_metadata:
         metadata.remove(cockpit_machines_metadata)
```

Comment:
I think this part is broken. Following the creation of VM, I run `virsh dumpxml vmName` both before and after I clicked on "Install", but in both cases the metadata seems untouched:
```
  <metadata xmlns:libosinfo="http://libosinfo.org/xmlns/libvirt/domain/1.0" xmlns:cockpit_machines="https://github.com/cockpit-project/cockpit-machines">
    <libosinfo:libosinfo>
      <libosinfo:os id="http://centos.org/centos-stream/9"/>
    </libosinfo:libosinfo>
    <cockpit_machines:data>
      <cockpit_machines:has_install_phase>true</cockpit_machines:has_install_phase>
      <cockpit_machines:install_source_type>os</cockpit_machines:install_source_type>
      <cockpit_machines:install_source/>
      <cockpit_machines:os_variant>centos-stream9</cockpit_machines:os_variant>
    </cockpit_machines:data>
  </metadata>
```

Reply:
I am not able to reproduce this. Before the "install" the metadata are expected to be there. After clicking install, the metadata are removed.

Reply:
Hmm that's strange, I checked again and for me it persists there. Does your metadata look the same before installation begins?

Reply:
Nope, I was wrong, this is actually a huge blocker, since if this metadata is left there, then after VM is installed, and you turn the VM off, UI will offer the option to install the VM again.

Reply:
Nope, I was wrong, this is actually a huge blocker, since if this metadata is left there, then after VM is installed, and you turn the VM off, UI will offer the option to install the VM again.

Reply:
@skobyda : I tested this: I created a VM as above, then clicked "Install". I tries to PXE boot (fails and keeps trying, of course), I do "force shut down". Then the state button changes to "Run", not "Install". This is even true after logging out and back in. Turns out I didn't look hard enough back then.. Right after "create and edit", the metadata has

      <cockpit_machines:has_install_phase>true</cockpit_machines:has_install_phase>

This is true even after logging out and in, or reboot. After "Run", it is still true. But after shutting down, it changes to

      <cockpit_machines:has_install_phase>false</cockpit_machines:has_install_phase>

which explains why we don't get the Install button again. I'll write an integration test for this to make sure it keeps working.

Do you mean you got the "Install" button again, or that it tries to do something different in the background?